tests/api_tests/users/test_views.py

```diff
@@ -437,11 +440,73 @@ def setUp(self):
             'researcherId': 'newResearcherId',
         }
 
+        self.missing_id = {
+            'type': 'users',
+            'fullname': 'el-Hajj Malik el-Shabazz',
+            'family_name': 'Z',
+        }
+
+        self.missing_type = {
+            'id': self.user_one._id,
+            'fullname': 'el-Hajj Malik el-Shabazz',
+            'family_name': 'Z',
+        }
+
+        self.incorrect_id = {
+            'id': '12345',
+            'type': 'users',
+            'fullname': 'el-Hajj Malik el-Shabazz',
+            'family_name': 'Z',
+        }
+
+        self.incorrect_type = {
+            'id': self.user_one._id,
+            'type': 'Wrong type.',
+            'fullname': 'el-Hajj Malik el-Shabazz',
+            'family_name': 'Z',
+        }
+
     def tearDown(self):
         super(TestUserUpdate, self).tearDown()
 
+    def test_patch_user_incorrect_type(self):
+        res = self.app.put_json_api(self.user_one_url, self.incorrect_type, auth=self.user_one.auth, expect_errors=True)
+        assert_equal(res.status_code, 409)
+
+    def test_patch_user_incorrect_id(self):
+        res = self.app.put_json_api(self.user_one_url, self.incorrect_id, auth=self.user_one.auth, expect_errors=True)
+        assert_equal(res.status_code, 409)
+
+    def test_patch_user_no_type(self):
+        res = self.app.put_json_api(self.user_one_url, self.missing_type, auth=self.user_one.auth, expect_errors=True)
+        assert_equal(res.status_code, 400)
+
+    def test_patch_user_no_id(self):
+        res = self.app.put_json_api(self.user_one_url, self.missing_id, auth=self.user_one.auth, expect_errors=True)
+        assert_equal(res.status_code, 400)
+
+    def test_partial_patch_user_incorrect_type(self):
+        res = self.app.patch_json_api(self.user_one_url, self.incorrect_type, auth=self.user_one.auth, expect_errors=True)
+        assert_equal(res.status_code, 409)
+
+    def test_partial_patch_user_incorrect_id(self):
+        res = self.app.patch_json_api(self.user_one_url, self.incorrect_id, auth=self.user_one.auth, expect_errors=True)
+        assert_equal(res.status_code, 409)
+
+    # PATCH does not require required fields.
+    def test_partial_patch_user_no_type(self):
+        res = self.app.patch_json_api(self.user_one_url, self.missing_type, auth=self.user_one.auth, expect_errors=True)
+        assert_equal(res.status_code, 200)
+
+    # PATCH does not require required fields.
```

Comment:
Better would be to write the test the way it should work, but put the skip decorator on it until you get the functionality in.